### ISSUE TITLE
GAModelWrapper fails if gradients are not defined

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#24]
When using GA in a project, I observed that when gradients failed to be defined in some part of the network, it would return a `ValueError`, whereas without GA it would give a warning.

We should try to match the behaviour of non-train-step-overloaded method in `Model.fit()`.

```
Traceback (most recent call last):
  File "main.py", line 103, in <module>
    main()
  File "main.py", line 97, in main
    trainer(ret)
  File "/home/andrep/workspace/bcgrade/milai/train.py", line 205, in trainer
    model.fit(
  File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/keras/utils/traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/tmp/__autograph_generated_fileisn3gvrb.py", line 15, in tf__train_function
    retval_ = ag__.converted_call(ag__.ld(step_function), (ag__.ld(self), ag__.ld(iterator)), None, fscope)
  File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/gradient_accumulator/GAModelWrapper.py", line 65, in train_step
    self.gradient_accumulation[i].assign_add(gradients[i])
ValueError: in user code:

    File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/keras/engine/training.py", line 1051, in train_function  *
        return step_function(self, iterator)
    File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/keras/engine/training.py", line 1040, in step_function  **
        outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/keras/engine/training.py", line 1030, in run_step  **
        outputs = model.train_step(data)
    File "/home/andrep/workspace/bcgrade/venv/lib/python3.8/site-packages/gradient_accumulator/GAModelWrapper.py", line 65, in train_step
        self.gradient_accumulation[i].assign_add(gradients[i])

    ValueError: None values not supported.
```

**Comment by andreped** [comment]
Managed to reproduce it. See [gist](https://colab.research.google.com/gist/andreped/f99974eb3188259f89c506a8abf833ef/fix_none_gradients.ipynb).

Also, found the fix [here](https://stackoverflow.com/a/72113077). Essentially doing this fixes it (gist has been updated to have this fix now):
```
tape.gradient(loss, weights, unconnected_gradients=tf.UnconnectedGradients.ZERO)
```

**Comment by andreped** [comment]
Fixed in d92e4fa13d0bd88b28fa4f8579d6e507eef65682
